src/components/tasks/list/taskListComponent.tsx

```diff
@@ -1,29 +1,29 @@
 import React from 'react';
-import TaskCard from '../single/taskCard';
 import { taskService } from '../../../services/taskService';
 import { TaskCardData } from '../../../models/task';
 import useAxiosEffect from '../../generic/useAxiosEffect';
 import ComponentStateHandler from '../../generic/componentStateHandler';
 import { useAppSelector } from '../../../context/hooks';
 import { currentUserData } from '../../../context/userReducer';
+import { splitTaskList } from './splitTaskList';
+import TaskCategoryComponent from './taskCategoryComponent';
+import CardsContainer from '../../styled-components/common/CardsContainer';
 
 const TaskListComponent = () => {
     const user = useAppSelector(currentUserData);
     const state = useAxiosEffect<TaskCardData[]>(taskService.getAll, [user.solved]);
     const tasks = state.data;
 
+    const categoryTasks = splitTaskList(tasks);
+
     return (
         <ComponentStateHandler state={state}>
-            <div className="flex flex-wrap w-9/12 sm:w-4/6 md:w-full xl:w-5/6 mx-auto">
-                {tasks &&
-                    tasks
-                        .sort((a, b) => a.ordinal - b.ordinal)
-                        .map((task) => (
-                            <div key={task.id} className="flex justify-center w-full md:w-1/2 lg:w-1/3 py-4 px-4">
-                                <TaskCard key={task.id} taskData={task} />
-                            </div>
-                        ))}
-            </div>
+            <CardsContainer>
+                {categoryTasks &&
```

Comment:
I think that `categoryTasks` cannot be undefined, because splitTaskList always returns an array.